### PR TITLE
added with credentials flag to the base command interface

### PR DESCRIPTION
**Pull request by FlorianTopf**
provided necessary parameter on request

@rokerkony I need `withCredentials` in "some" project, so I'd be glad to have the flag as part of the interface for the apiCommand.

### CODE REVIEW

File: src/url-search-params.builder.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ export class UrlSearchParamsBuilder {
             queryParameters = [];
         }
         const params: URLSearchParams = new URLSearchParams();
-        queryParameters.forEach((searchParameter: QueryParameter): void => {
+        queryParameters.forEach((searchParameter: [string, string | number]): void => {
```
Comment (rokerkony) [inline src/url-search-params.builder.ts]: why this change?

Comment (rokerkony) [inline src/url-search-params.builder.ts]: I got it ... please remove it from this PR... I pushed it to `refactoring/change-base-command-to-interface-and-separate-url-construction`

Comment (FlorianTopf) [inline src/url-search-params.builder.ts]: when I did npm install on your refactoring branch the linter was complaining about an non existing type. This type is not existing.

**Comment by FlorianTopf** [comment]
@rokerkony just tested it on my app. :1st_place_medal:

**Comment by rokerkony** [comment]
@FlorianTopf could you please rebase with master ?
